app/services/sign_in/user_loader.rb

```diff
@@ -40,6 +41,17 @@ def validate_account_and_session
       raise Errors::SessionNotFoundError.new message: 'Invalid Session Handle' unless session
     end
 
+    def validate_mpi_profile
+      profile_errors = current_user.validate_mpi_profile
+      return if profile_errors.empty?
+
+      if profile_errors.include?(:mpi_death_flag)
```

Comment:
Is there any reason the user `validate_mpi_profile` couldn't raise these errors instead of in here?